packages/page-broker/src/Overview/Workload.tsx

```diff
@@ -3,76 +3,105 @@
 
 import type { ApiPromise } from '@polkadot/api';
 import type { RegionInfo } from '@polkadot/react-hooks/types';
+import type { Option } from '@polkadot/types';
+import type { PalletBrokerStatusRecord } from '@polkadot/types/lookup';
 import type { CoreWorkloadType, CoreWorkplanType, InfoRow } from '../types.js';
 
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
 import { ExpandButton } from '@polkadot/react-components';
-import { useRegions, useToggle } from '@polkadot/react-hooks';
+import { useApi, useBrokerSalesInfo, useCall, useRegions, useToggle } from '@polkadot/react-hooks';
 
 import { formatRowInfo } from '../utils.js';
 import WorkInfoRow from './WorkInfoRow.js';
 import Workplan from './Workplan.js';
 
 interface Props {
   api: ApiPromise;
-  value: CoreWorkloadType
-  timeslice: number;
-  workplan?: CoreWorkplanType[] | null
+  core: number;
+  workload: CoreWorkloadType[] | undefined
+  workplan?: CoreWorkplanType[] | undefined
 }
 
-function Workload ({ api, timeslice, value: { core, info, lastBlock, type }, workplan }: Props): React.ReactElement<Props> {
+function Workload ({ api, core, workload, workplan }: Props): React.ReactElement<Props> {
+  const { isApiReady } = useApi();
+  const salesInfo = useBrokerSalesInfo();
+
+  const status = useCall<Option<PalletBrokerStatusRecord>>(isApiReady && api.query.broker?.status);
   const [isExpanded, toggleIsExpanded] = useToggle(false);
-  const [tableData, setTableData] = useState<InfoRow>();
-  const [currentRegion, setCurrentRegion] = useState<RegionInfo | undefined>();
+  const [workloadData, setWorkloadData] = useState<InfoRow[]>();
+  const [workplanData, setWorkplanData] = useState<InfoRow[]>();
+
+  const currentTimeSlice = useMemo(() => {
+    if (!!status && !!status?.toHuman()) {
+      return status.unwrap().lastCommittedTimeslice.toNumber();
+    }
+
+    return 0;
+  }, [status]);
+
   const regionInfo = useRegions(api);
+  const region: RegionInfo | undefined = useMemo(() => regionInfo?.find((v) => v.core === core && v.start <= currentTimeSlice && v.end > currentTimeSlice), [regionInfo, core, currentTimeSlice]);
 
   useEffect(() => {
-    if (info) {
-      const region: RegionInfo | undefined = regionInfo?.find((v) => v.core === core && v.start <= timeslice && v.end > timeslice);
+    if (!!workload?.length && !!salesInfo) {
```

Comment:
```suggestion
    if (workload?.length && salesInfo) {
```

I think that would work just fine. Im guessing here you just want to ensure that if length is greater than 0?